d2/src/main/java/com/linkedin/d2/xds/XdsClient.java

```diff
@@ -95,6 +95,23 @@ final void onChanged(ResourceUpdate update)
     }
   }
 
+  public static abstract class D2UriResourceWatcher
```

Comment:
we should make this extends ResourceWatcher for better code structure, so that:
1. watchD2Uri is not needed, and can just reuse watchXdsResource(globCollectionUrn(cluster, uri), watcher).
2. D2UriSubscriber is not needed, and can just use ResourceSubscriber.

Reply:
Sure, makes sesne